### ISSUE TITLE
Is fake_spec.detach() in Generator training step intentional?

### ISSUE THREAD

**Issue by kirak-kim** [nikhilsinghmus/room2reverb#37]
Hello,

First of all, thank you for this interesting work!

I have a question about the Generator training step in `training_step()` and wanted to check whether the current behavior is intentional before assuming it's a bug.

When training the Generator (`optimizer_idx == 0 or 1`), `fake_spec.detach()` is passed to the Discriminator (line 102 in model.py):

```python
# training_step(), optimizer_idx == 0 or 1
d_fake2 = self.d(**fake_spec.detach()**, f)  # is .detach() here intentional?
G_loss1 = F.mse_loss(d_fake2, torch.ones(d_fake2.shape, device=self.device))
```

My understanding is that '.detach()' here would cut the gradient path from 'd_fake2' back to 'fake_spec', meaning 'G_loss1' would not propagate gradients into the Generator, leaving only 'G_loss2' to effectively update the Generator weights.

When I trained on my own dataset, I observed the Discriminator dominating very early in training (D_loss ~0.2, D_real_mean ~0.9), which seemed consistent with this behavior.

I wondered if removing `.detach()` here might be the intended fix.

Thank you!

**Comment by nikhilsinghmus** [comment]
Hi @kirak-kim,

Thank you so much for your detailed comment. You're right, that `.detach()` should not be there for the generator step and is a bug. I unfortunately don't have a reliable way to verify whether this was present in the original training runs, but to my memory it was not. It's not intentional and is likely just a typo; there were several changes made in an environment that no longer exists. I should patch this in the codebase along with a few other things and will plan to do so.

**Comment by kirak-kim** [comment]
Thank you for the clarification!
